sys/newlib_syscalls_default/syscalls.c

```diff
@@ -56,12 +56,70 @@
 #include "xtimer.h"
 #endif
 
+#ifndef NUM_HEAPS
+#define NUM_HEAPS 1
+#endif
+
 /**
  * @brief manage the heap
  */
 extern char _sheap;                 /* start of the heap */
 extern char _eheap;                 /* end of the heap */
-char *heap_top = &_sheap + 4;
+
+extern char _sheap1;
+extern char _eheap1;
+
+extern char _sheap2;
+extern char _eheap2;
+
+extern char _sheap3;
+extern char _eheap3;
```

Comment:
Hmm, maybe it makes sense to add a comment that we rely on the compiler to not generate references to the symbols `_sheap<num>` and `_eheap<num>` in case they are not used; so there is no need to provide these symbols unless the heaps are actually used.

Reply:
I added a comment section.